matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -85,20 +85,137 @@ async fn signalling_loop<S: Signaller>(
     }
 }
 
+/// The raw format of data being sent and received.
+pub type Packet = Box<[u8]>;
+
+trait DataChannel {
+    fn send(&mut self, packet: Packet) -> Result<(), MessagingError>;
+}
+
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 trait Messenger {
-    async fn message_loop(
-        id_tx: Sender<PeerId>,
-        config: WebRtcSocketConfig,
-        channels: MessageLoopChannels,
-    );
+    type DataChannel: DataChannel;
+    type PeerLoopArgs: Send;
+
+    async fn offer_handshake(
+        signal_peer: SignalPeer,
+        from_peer_rx: UnboundedReceiver<PeerSignal>,
+        peer_state_tx: UnboundedSender<(PeerId, PeerState)>,
+        messages_from_peers_tx: Vec<UnboundedSender<(PeerId, Packet)>>,
+        config: &WebRtcSocketConfig,
+    ) -> (PeerId, Vec<Self::DataChannel>, Self::PeerLoopArgs);
```

Comment:
Another thought: Maybe it's better to have this return a struct `HandshakeResult<T>` with a platform-specific field?

Sketch:

```rust
struct HandshakeResult<T> {
    id: PeerId,
    data_senders: Self::PeerDataSender,
    platform_specific: T
}
```

Reply:
Yeah, that's probably the way to go, definitely more readable